ui/react-components/components/EditAppointment/EditAppointment.jsx

```diff
@@ -362,15 +502,30 @@ const EditAppointment = props => {
                     <AppointmentNotes value={appointmentDetails.notes} onChange={(event) => updateAppointmentDetails({notes: event.target.value})}/>
                 </div>
             </div>
-            <AppointmentEditorFooter checkAndSave={!isRecurringAppointment() ? checkAndSave : undefined} isEdit={true} showUpdateOptions={showUpdateOptions} />
+            <AppointmentEditorFooter
+                checkAndSave={applyForAllInd => {
+                    if (isRecurringAppointment()) {
+                        if (applyForAllInd === undefined) {
+                            applyForAllInd = !isStartDateModified();
+                        }
+                        setApplyForAll(applyForAllInd);
+                        checkAndUpdateRecurringAppointments(applyForAllInd).then();
+                    } else {
+                        checkAndSave().then();
+                    }
+                }}
```

Comment:
Extracting it as a method makes it more readable

Reply:
DONE